### ISSUE TITLE
hdf5storage 0.1.18 is broken with setuptools 66.0.0

### ISSUE THREAD

**Issue by joerg-tt** [frejanordsiek/hdf5storage#125]
As of https://github.com/pypa/setuptools/commit/217d9267e017aed6ddff36fa47d82ceb475cbf22 version 66.0.0, hdf5storage fails to initialize the MarshallerCollection.

Reason: In https://github.com/frejanordsiek/hdf5storage/blob/0.1.18/hdf5storage/__init__.py#L902 one of the elements loaded from the Marshallers module is `parse_version`, which now is a class in setuptools instead of a function. Subsequently, the call `m()` fails, because `parse_version` expects an argument.

![grafik](https://user-images.githubusercontent.com/69461468/212936370-7bf50de2-5cf9-4bfe-bbc9-51143b36e43d.png)

```
  File "C:\Python310\lib\site-packages\hdf5storage\__init__.py", line 1759, in loadmat
    options = Options(marshaller_collection=marshaller_collection)
  File "C:\Python310\lib\site-packages\hdf5storage\__init__.py", line 260, in __init__
    self.marshaller_collection = MarshallerCollection()
  File "C:\Python310\lib\site-packages\hdf5storage\__init__.py", line 902, in __init__
    self._builtin_marshallers = [m() for key, m in dict(
  File "C:\Python310\lib\site-packages\hdf5storage\__init__.py", line 902, in <listcomp>
    self._builtin_marshallers = [m() for key, m in dict(
TypeError: Version.__init__() missing 1 required positional argument: 'version'
```

Curiously, this already seems to be fixed in the main branch: https://github.com/frejanordsiek/hdf5storage/blob/main/hdf5storage/__init__.py#L1115

Would it be possible to have a new hdf5storage release that includes this fix?

Thanks a lot!

**Comment by frejanordsiek** [comment]
It is now fixed in commit b223ff7212c12b9e7ad920906e2faf3081e06881. It was caused by very primitive collection of the Marshallers (collecting everything that is a class)

I will be pushing a 0.1.19 bugfix release to PyPI shortly.

**Comment by joerg-tt** [comment]
Thanks!
